File: tests/resources/test_codegen/template_jinja.swift
import Foundation

{% if mod %}
let MOD = {{ mod }}
{% endif %}
{% if yes_str %}
let YES = "{{ yes_str }}"
{% endif %}
{% if no_str %}
let NO = "{{ no_str }}"
{% endif %}
{% if prediction_success %}

func solve({{ formal_arguments }}) {
    {% if yes_str %}
    var ans = false

    print(ans ? YES : NO)
    {% else %}
    var ans = 0

    print(ans)
    {% endif %}
}
{% endif %}

func main() {
    {% if prediction_success %}
    var tokenIndex = 0, tokenBuffer = [String]()
    func readString() -> String {
        if tokenIndex >= tokenBuffer.count {
            tokenIndex = 0
            tokenBuffer = readLine()!.split(separator: " ").map { String($0) }
        }
        defer { tokenIndex += 1 }
        return tokenBuffer[tokenIndex]
    }
    func readInt() -> Int { Int(readString())! }
    func readDouble() -> Double { Double(readString())! }
    {{input_part}}
    _ = solve({{ actual_arguments }})
    {% else %}
    // Failed to predict input format
    {% endif %}
}

#if DEBUG
let caseNumber = 1
_ = freopen("in_\(caseNumber).txt", "r", stdin)
#endif

main()

File: tests/resources/test_codegen/template_jinja.cpp
#include <iostream>
#include <sstream>
#include <fstream>
#include <string>
#include <vector>
#include <deque>
#include <queue>
#include <stack>
#include <set>
#include <map>
#include <algorithm>
#include <functional>
#include <utility>
#include <bitset>
#include <cmath>
#include <cstdlib>
#include <ctime>
#include <cstdio>
#include <cassert>
using namespace std;

{% if mod is not none %}
const int mod = {{ mod }};
{% endif %}
{% if yes_str is not none %}
const string YES = "{{ yes_str }}";
{% endif %}
{% if no_str is not none %}
const string NO = "{{ no_str }}";
{% endif %}
void solve({{ formal_arguments }}){

}
int main(){
    {{input_part}}
    solve({{ actual_arguments }});
    return 0;
}
